tutorials/2d/custom_drawing_in_2d.rst

```diff
@@ -195,333 +209,850 @@ The same applies to the ``draw_rect`` method with ``filled = false``.
+    for high resolution screens (larger than 1080p). If that is your case, and the
+    drawing is too small consider increasing your window scale in
+    ``Menu > Project > Project settings > display/window/stretch/scale`` to adjust
+    the project to a higher resolution (a 2 or 4 scale tends to work well).
 
-Basically, drawing a shape on the screen requires it to be decomposed into a certain number of points
-linked from one to the next. As you can imagine, the more points your shape is made of,
-the smoother it will appear, but the heavier it will also be in terms of processing cost. In general,
-if your shape is huge (or in 3D, close to the camera), it will require more points to be drawn without
-it being angular-looking. On the contrary, if your shape is small (or in 3D, far from the camera),
-you may decrease its number of points to save processing costs; this is known as *Level of Detail (LOD)*.
-In our example, we will simply use a fixed number of points, no matter the radius.
+Drawing a custom polygon shape
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While there is a dedicated node to draw custom polygons (
+:ref:`Polygon2D <class_Polygon2D>`), we will use in this case exclusively lower
+level drawing functions to combine them on the same node and be able to create
+more complex shapes later on.
+
+First, we will define a set of points -or X and Y coordinates- that will form
+the base of our shape:
 
 .. tabs::
  .. code-tab:: gdscript GDScript
 
-    func draw_circle_arc(center, radius, angle_from, angle_to, color):
-        var nb_points = 32
-        var points_arc = PackedVector2Array()
-
-        for i in range(nb_points + 1):
-            var angle_point = deg_to_rad(angle_from + i * (angle_to-angle_from) / nb_points - 90)
-            points_arc.push_back(center + Vector2(cos(angle_point), sin(angle_point)) * radius)
+    extends Node2D
 
-        for index_point in range(nb_points):
-            draw_line(points_arc[index_point], points_arc[index_point + 1], color)
+    var coords_head : Array = [
+        [ 22.952, 83.271 ],  [ 28.385, 98.623 ],
+        [ 53.168, 107.647 ], [ 72.998, 107.647 ],
+        [ 99.546, 98.623 ],  [ 105.048, 83.271 ],
+        [ 105.029, 55.237 ], [ 110.740, 47.082 ],
+        [ 102.364, 36.104 ], [ 94.050, 40.940 ],
+        [ 85.189, 34.445 ],  [ 85.963, 24.194 ],
+        [ 73.507, 19.930 ],  [ 68.883, 28.936 ],
+        [ 59.118, 28.936 ],  [ 54.494, 19.930 ],
+        [ 42.039, 24.194 ],  [ 42.814, 34.445 ],
+        [ 33.951, 40.940 ],  [ 25.637, 36.104 ],
+        [ 17.262, 47.082 ],  [ 22.973, 55.237 ]
+    ]
 
  .. code-tab:: csharp
 
-    public void DrawCircleArc(Vector2 center, float radius, float angleFrom, float angleTo, Color color)
-    {
-        int nbPoints = 32;
-        var pointsArc = new Vector2[nbPoints + 1];
+    using Godot;
 
-        for (int i = 0; i <= nbPoints; i++)
+    public partial class MyNode2D : Node2D
+    {
+        private float[,] coordsHead =
```

Comment:
The C# naming convention for private fields is camelCase prefixed by an underscore.

```suggestion
        private float[,] _coordsHead =
```